test_regress/driver.pl

```diff
@@ -1847,6 +1857,15 @@ sub _try_regex {
     return $result;
 }
 
+sub _make_vpi_main {
+    my $self = shift;
+    $self->_read_inputs_v();
+    my $filename = $self->{main_filename};
+    my $fh = IO::File->new(">$filename") or die "%Error: $! $filename,";
+    my $verilator_cpp = 't_vpi_main.cpp';
+    File::Copy::copy($verilator_cpp, $filename) or die "Copy failed: $!";
+}
+
```

Comment:
I'd suggest rename t_vpi_main.cpp to e.g. TestVpiMain.cpp since it is now a common file.

What prevents just putting TestVpiMain.cpp onto the verilog_flags2 command line in the .pl file, and so not needing these driver changes?

Reply:
So in order to get that working I'd need to have vm_prefix also override self->vm_prefix, but this will break all of the t_trace_two tests. So should I add a different param for prefixing? I prefer the current implementation because it does both actions for you, but it could do without file copying

Reply:
Sorry, I don't understand your point.

Reply:
I'd really prefer to not do so much driver.pl plumbing, I barely know perl

Reply:
I don't understand why you cannot leave driver.pl without any changes and add a 

```
verilator_flags2=>['TseVpiMain.cpp'],
```

to the test.pl.


I didn't understand your reply about vm prefix.

Reply:
TestVpiMain.cpp uses Vtop as the top module name. Oh I see now I could've used the VM_PREFIX_INCLUDE convention. Cocotb just uses --prefix Vtop for every build, which I guess using here actually complicates things more. I think if verilator tests switched to vtop it would be possible to remove a lot of code from driver.pl

Reply:
Right, please use VM_PREFIX/VM_PREFIX_INCLUDE, then backout driver.pl and think we're ready.

I don't want to switch to Vtop as prefer to have tests rely on the default behavior. Maybe once C++20 is required and --main can be relied on will revisit that choice.